pkg/metadata/provider_scaleway.go

```diff
@@ -220,7 +220,7 @@ func scalewayGetUserdata() ([]byte, error) {
 	defer resp.Body.Close()
 
 	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
+	if err != nil || resp.StatusCode == 404 {
```

Comment:
That's weird. First the `Extract` method (the one you are using I guess) does not return any errors for failed userdata. 
Second, if I understood correctly, you wanted to treat 404 as no error, but here you return an error. I would check the error code here https://github.com/linuxkit/linuxkit/pull/3429/files#diff-1b03b5d41b65159de69dd1b6ea7a31daR217 and return no error if 404

Reply:
I'll move this check up to the previous error check, it was put here because I first thought the ststuscode was on the body. I'll also fix the check as this is not working as expected as you said.

Reply:
I moved the error check [here](https://github.com/linuxkit/linuxkit/pull/3429/files#diff-1b03b5d41b65159de69dd1b6ea7a31daR217), as you said.

However, my intension was to actually make 404 responses return an error as letting it through will return `"Invalid Key"` as the userdata because that's the body of the response in the case of a 404.

This should be ok though, as the `Extract` method catches all errors and treats them as non errors [here](https://github.com/linuxkit/linuxkit/pull/3429/files#diff-1b03b5d41b65159de69dd1b6ea7a31daR143-R150)